src/python/zquantum/core/interfaces/backend.py

```diff
@@ -47,46 +57,65 @@ def run_circuitset_and_measure(
         Returns:
             Measurements for each circuit.
         """
-        measurement_set = []
-        for circuit in circuit_set:
-            measurement_set.append(self.run_circuit_and_measure(circuit), **kwargs)
-        return measurement_set
+        if not self.supports_batching:
+            measurement_set = []
+            for circuit in circuit_set:
+                measurement_set.append(self.run_circuit_and_measure(circuit), **kwargs)
+            return measurement_set
+        else:
+            self.number_of_circuits_run += len(circuit_set)
+            self.number_of_jobs_run += 1
 
     def get_expectation_values(
-        self, circuit: Circuit, qubit_operator: QubitOperator, **kwargs
+        self, circuit: Circuit, operator: SymbolicOperator, **kwargs
     ) -> ExpectationValues:
         """
         Executes the circuit and calculates the expectation values for given operator.
 
         Args:
             circuit (core.circuit.Circuit): quantum circuit to be executed.
-            qubit_operator(openfermion): Operator for which we calculate the expectation value.
+            operator(openfermion.IsingOperator): Operator for which we calculate the expectation value.
 
         Returns:
             ExpectationValues: object representing expectation values for given operator.
         """
-        raise NotImplementedError
+        measurements = self.run_circuit_and_measure(circuit)
+        operator = change_operator_type(operator, IsingOperator)
+        expectation_values = measurements.get_expectation_values(operator)
+        expectation_values = expectation_values_to_real(expectation_values)
+        return expectation_values
 
     def get_expectation_values_for_circuitset(
-        self, circuitset: List[Circuit], qubit_operator: QubitOperator, **kwargs
-    ) -> [ExpectationValues]:
+        self, circuitset: List[Circuit], operator: IsingOperator, **kwargs
```

Comment:
You've changed it to `SymbolicOperator` in `get_expectation_values` but here it's `IsingOperator`. 
Please make it consistent.

Reply:
Addressed here: 9f87e46